src/grandorgue/sound/GOSoundFilter.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2006 Milan Digital Audio LLC
+ * Copyright 2009-2024 GrandOrgue contributors (see AUTHORS)
+ * License GPL-2.0 or later
+ * (https://www.gnu.org/licenses/old-licenses/gpl-2.0.html).
+ */
+
+#ifndef GOSOUNDFILTER_H_
+#define GOSOUNDFILTER_H_
+
+#include <cmath>
+
+enum class GO_FILTER_TYPE { LPF, HPF, LOW_SHELF, HIGH_SHELF };
```

Comment:
What's your opinion on the usage of the more modern, strongly type safe, `enum class` vs a plain `enum`? Mostly I'm concerned about the `enum class` needing the `static_cast<bool>(m_type)` in the evaluation vs the plain `enum` could be evaluated directly.

Reply:
enum class allows to specify the basic type, ie. `uint8_t`. So I'd prefer enum class rather than enum.

Reply:
@oleg68 I was thinking mostly about the necessary `static_cast<bool>(m_type)` and if you think that's problematic for performance or not? When a separate bool was used the comparison would be fast as would a plain `enum` that implicitly would be converted so that a 0 value would equal to a bool value of false and any other to true. The `enum class` that I choose to use won't allow that without an explicit conversion. Your clever idea of using the filter none as a zero value for comparison would work without any cast for a plain `enum,` but needs to be explicit with `enum class`.

If you think this is of no consequence, I'm not bothered either.